src/Process/PlaceOnGroundProcessing.js

```diff
@@ -0,0 +1,131 @@
+import * as THREE from 'three';
+import Coordinates from '../Core/Geographic/Coordinates';
+import Extent from '../Core/Geographic/Extent';
+import RendererConstant from '../Renderer/RendererConstant';
+import { l_ELEVATION } from '../Renderer/LayeredMaterial';
+
+function tileAt(pt, tile) {
+    if (tile.extent) {
+        if (!tile.extent.pointIsInside(pt)) {
+            return undefined;
+        }
+
+        for (const c of tile.children) {
+            const t = tileAt(pt, c);
+            if (t) {
+                return t;
+            }
+        }
+        if (tile.getLayerTextures(l_ELEVATION)[0].coords.zoom > -1) {
+            return tile;
+        }
+        return undefined;
+    }
+}
+
+let _canvas;
+function _readTextureValueAt(layer, texture, x, y) {
+    if (texture.image.data) {
+        return texture.image.data[y * texture.image.width + x];
+    } else {
+        if (!_canvas) {
+            _canvas = document.createElement('canvas');
+            _canvas.width = 1;
+            _canvas.height = 1;
+        }
+        const ctx = _canvas.getContext('2d');
+        ctx.drawImage(texture.image, x, y, 1, 1, 0, 0, 1, 1);
+        const d = ctx.getImageData(0, 0, 1, 1);
+        return THREE.Math.lerp(
+            layer.materialOptions.colorTextureElevationMinZ,
+            layer.materialOptions.colorTextureElevationMaxZ,
+            d.data[0] / 255);
+    }
+}
+
+function _updateSinglePoint(referenceCrs, layer, nodes, crs, toModify, zOffset) {
```

Comment:
It would seem clearer to me, to move the texture reading logic in material or Tile. `PlaceOnGroundProcessing.js` would contain only the clamping logic.
```js
if (tileWithValidElevationTexture) {
        const z = tileWithValidElevationTexture.material.getElevation(pt);
        // or 
        const z = tileWithValidElevationTexture.getElevation(pt);

        pt._values[2] = z + zOffset;

        toModify.copy(pt.as(referenceCrs).xyz());
        return tileWithValidElevationTexture;
}
```
In addition, the `getElevation(coords)` function can also be used for `pickAltitude`.